components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreManagementServiceImpl.java

```diff
@@ -0,0 +1,316 @@
+        return keyStoreData;
+    }
+
+    private List<String> getAliasList(KeyStoreData keyStoreData) {
+
+        List<String> aliasList = new ArrayList<>();
+        CertData key = keyStoreData.getKey();
+        if (key != null && key.getAlias() != null) {
+            aliasList.add(key.getAlias());
+        }
+
+        CertData[] certDataArray = keyStoreData.getCerts();
+        for (CertData certData : certDataArray) {
+            String alias = certData.getAlias();
+            if (alias != null) {
+                aliasList.add(alias);
+            }
+        }
+        return aliasList;
+    }
+
+    private List<String> filterAlias(List<String> aliases, String filter) throws KeyStoreManagementException {
+
+        if (filter != null) {
+            filter = filter.replace(" ", "+");
+            String[] extractedFilter = filter.split("[+]");
+            if (extractedFilter.length == 3) {
+                if (StringUtils.equals(extractedFilter[0], FILTER_FIELD_ALIAS)) {
+                    String operation = extractedFilter[1];
+                    String value = extractedFilter[2];
+                    if (StringUtils.equals(operation, FILTER_OPERATION_EQUALS)) {
+                        aliases = aliases.stream().filter(alias -> alias.matches(value))
+                                .collect(Collectors.toList());
+                    } else if (StringUtils.equals(operation, FILTER_OPERATION_STARTS_WITH)) {
+                        aliases = aliases.stream().filter(alias -> alias.startsWith(value))
+                                .collect(Collectors.toList());
+                    } else if (StringUtils.equals(operation, FILTER_OPERATION_ENDS_WITH)) {
+                        aliases = aliases.stream().filter(alias -> alias.endsWith(value))
+                                .collect(Collectors.toList());
+                    } else if (StringUtils.equals(operation, FILTER_OPERATION_CONTAINS)) {
+                        aliases = aliases.stream().filter(alias -> alias.contains(value))
+                                .collect(Collectors.toList());
+                    } else {
+                        handleClientException(ERROR_CODE_UNSUPPORTED_FILTER_OPERATION, operation);
+                    }
+                }
+            } else {
+                handleClientException(ERROR_CODE_BAD_VALUE_FOR_FILTER, filter);
+            }
+        }
+        return aliases;
+    }
+
+    private String getTenantDomain() {
+
+        return CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+    }
+
+    private boolean isSuperTenant() {
+
+        return CarbonContext.getThreadLocalCarbonContext().getTenantId() == MultitenantConstants.SUPER_TENANT_ID;
+    }
+
+    private String getTrustStoreName() {
+
+        ServerConfiguration serverConfiguration = ServerConfiguration.getInstance();
+        String fileName = serverConfiguration.getFirstProperty(SERVER_TRUSTSTORE_FILE);
+        int index = fileName.lastIndexOf('/');
```

Comment:
We can use the Java Path API here to get the file name :)

Reply:
https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#getFileName()

Reply:
fixed